### PR TITLE
Trainer het

### PR DESCRIPTION
**Pull request by caijie1998**
add trainer (including performance test)

### CODE REVIEW

File: autogl/module/train/node_classification_het.py [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+            dict(
+                [
+                    x
+                    for x in hp.items()
+                    if x[0] in [y["parameterName"] for y in model.space]
+                ]
+            )
+        )
+
+        ret = self.__class__(
```
Comment (Frozenmad) [inline autogl/module/train/node_classification_het.py:443]: The initialization process seems not the same with the one in ` __init__` func?

File: test/performance/heterogeneous/dgl/han_base/base.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        if early_stop:
+            break
+
+    stopper.load_checkpoint(model)
+    test_loss, test_acc, test_micro_f1, test_macro_f1 = evaluate(model, g, features, labels, test_mask, loss_fcn)
+    print('Test loss {:.4f} | Test Micro f1 {:.4f} | Test Macro f1 {:.4f}'.format(
+        test_loss.item(), test_micro_f1, test_macro_f1))
+
+if __name__ == '__main__':
+    import argparse
```
Comment (Frozenmad) [inline test/performance/heterogeneous/dgl/han_base/base.py:93]: Can we add a loop outside? So that we can see the mean and std of each metric?

File: test/performance/heterogeneous/dgl/han_main.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+import os
+import datetime
+import errno
+import numpy as np
+from tqdm import tqdm
+import dgl
+os.environ["AUTOGL_BACKEND"] = "dgl"
+import sys
+import os.path as osp
+sys.path.append("../../../../")
```
Comment (Frozenmad) [inline test/performance/heterogeneous/dgl/han_main.py:13]: these lines should be resolved if you install the local library through `pip install -e .`

File: test/performance/heterogeneous/dgl/han_main.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    parser.add_argument('--lr', type=float, default=0.005,
+                        help='Learning rate')
+    args = parser.parse_args().__dict__
+    args['dataset'] = 'ACMRaw' if not args['hetero'] else 'ACM'
+    args['device'] = 'cuda:6' if torch.cuda.is_available() else 'cpu'
+    args['num_epochs'] = 10
+    set_random_seed(args['seed'])
+    print(args)
+
+    main(args)
```
Comment (Frozenmad) [inline test/performance/heterogeneous/dgl/han_main.py:309]: Can we add a loop?

**Review by Frozenmad — COMMENTED** [review]
Overall very excellent work on hetero model and trainer!

There remain some general issues:
1. Do not include the dataset in the commit. The code repository should not include any binary files because it is not a cloud storage. We should make it nice and clean. If any BIG file is included, the git will become larger and larger in the following developments. Please see [here](https://www.zhihu.com/question/54419234) to remove all the related commits related with the `ACM.mat` file.
2. Please keep in mind that, except for the forward of model, all the datasets we handle is and only is a DGLDataset. Please be aware that all the masks, features, key fields should be derived from the DGLDataset.
3. Currently the trainer should not support pyg backend. So you need to `delete` the import of hgt trainer if the backend if pyg.

Other comments are listed below.

File: autogl/module/train/__init__.py [hunk 5]
```diff
@@ -26,6 +26,9 @@ def register_trainer_cls(cls):
 
 from .graph_classification_full import GraphClassificationFullTrainer
 from .node_classification_full import NodeClassificationFullTrainer
+from .node_classification_hgt import NodeClassificationHgtTrainer
+from .node_classification_han import NodeClassificationHanTrainer
+from .node_classification_het import NodeClassificationHetTrainer
```
[the remark this answers is not in the document]
Comment (caijie1998) [reply autogl/module/train/__init__.py:29]: > I have delete the extra node_classiFICation_hgt and node_classiFICation_han (I used them for previous test).In fact there is only one uniform NodeClassificationHetTrainer for hgt and han.

File: autogl/module/train/base.py [hunk 6]
```diff
@@ -420,3 +400,70 @@ def __init__(
...
+        else:
+            raise TypeError(
+                f"Model argument only support str or BaseModel, got {model}."
+            )
+        super(_BaseClassificationHetTrainer, self).__init__(
+            _model, __device, init, feval, loss
+        )
+
+
+class BaseNodeClassificationHetTrainer(_BaseClassificationHetTrainer):
```
[the remark this answers is not in the document]
Comment (caijie1998) [reply autogl/module/train/base.py:451]: 很有道理. I directly delete current BaseNodeClassificationHetTrainer and change the name of _BaseClassificationHetTrainer to BaseNodeClassificationHetTrainer. And I have tested after the adjustment.

File: autogl/module/train/__init__.py [hunk 7]
```diff
@@ -26,6 +26,7 @@ def register_trainer_cls(cls):
 
 from .graph_classification_full import GraphClassificationFullTrainer
 from .node_classification_full import NodeClassificationFullTrainer
+from .node_classification_het import NodeClassificationHetTrainer
```
Comment (Frozenmad) [inline autogl/module/train/__init__.py:29]: You need to put NodeClassificationHetTrainer keyword in the `__all__` list in this file.

File: autogl/module/train/base.py [hunk 8]
```diff
@@ -420,3 +400,70 @@ def __init__(
...
+        else:
+            raise TypeError(
+                f"Model argument only support str or BaseModel, got {model}."
+            )
+        super(BaseNodeClassificationHetTrainer, self).__init__(
+            _model, __device, init, feval, loss
+        )
+
+
+class BaseNodeClassificationHetTrainer(_BaseClassificationHetTrainer):
```
Comment (Frozenmad) [inline autogl/module/train/base.py:451]: Please delete this class, or it will overwrite the class above

**Comment by Frozenmad** [comment]
Why would our model need meta_path in the `__init__`? Can't we extract meta_path from dataset?
